editTemplateView.xhtml: compare doesn't work on wildfly 12.0

Exception:
```
2018-05-23 10:40:58,127 logLev=SEVERE class=AMWExceptionHandler thread="default task-6" corrID= caller= user= An Error occured: javax.faces.component.UpdateModelException: javax.el.ELException: /pages/editTemplateView.xhtml @14,76 value="#{editTemplateView.contextIdViewParam}": java.lang.NullPointerException
        at javax.faces.component.UIInput.updateModel(UIInput.java:868)
        at javax.faces.component.UIViewParameter.updateModel(UIViewParameter.java:331)
        at javax.faces.component.UIInput.processUpdates(UIInput.java:751)
        at javax.faces.component.UIComponentBase.processUpdates(UIComponentBase.java:1291)
        at javax.faces.component.UIComponentBase.processUpdates(UIComponentBase.java:1291)
        at javax.faces.component.UIViewRoot.processUpdates(UIViewRoot.java:1254)
        at com.sun.faces.lifecycle.UpdateModelValuesPhase.execute(UpdateModelValuesPhase.java:78)
        at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
        at com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:198)
        at javax.faces.webapp.FacesServlet.service(FacesServlet.java:658)
        at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
        at ch.puzzle.itc.mobiliar.presentation.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:43)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at ch.puzzle.itc.mobiliar.presentation.util.CachingFilter.doFilter(CachingFilter.java:65)
        at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
        at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
        at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
        at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
        at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:67)
        at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
        at org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)
        at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
        at io.undertow.server.handlers.DisableCacheHandler.handleRequest(DisableCacheHandler.java:33)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:53)
        at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
        at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
        at io.undertow.servlet.handlers.security.ServletSecurityConstraintHandler.handleRequest(ServletSecurityConstraintHandler.java:59)
        at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
        at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
        at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
        at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at org.wildfly.extension.undertow.deployment.GlobalRequestControllerHandler.handleRequest(GlobalRequestControllerHandler.java:68)
        at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
        at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292)
        at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81)
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138)
        at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135)
        at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
        at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
        at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
        at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1526)
        at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1526)
        at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1526)
        at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1526)
        at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1526)
        at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272)
        at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)
        at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:104)
        at io.undertow.server.Connectors.executeRootHandler(Connectors.java:360)
        at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1985)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1487)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1378)
        at java.lang.Thread.run(Thread.java:748) [rt.jar:1.8.0_151]
Caused by: javax.el.ELException: /pages/editTemplateView.xhtml @14,76 value="#{editTemplateView.contextIdViewParam}": java.lang.NullPointerException
        at com.sun.faces.facelets.el.TagValueExpression.setValue(TagValueExpression.java:139)
        at javax.faces.component.UIInput.updateModel(UIInput.java:834)
        ... 62 more
Caused by: javax.el.ELException: java.lang.NullPointerException
        at javax.el.BeanELResolver.setValue(BeanELResolver.java:325)
        at com.sun.faces.el.DemuxCompositeELResolver._setValue(DemuxCompositeELResolver.java:255)
        at com.sun.faces.el.DemuxCompositeELResolver.setValue(DemuxCompositeELResolver.java:281)
        at com.sun.el.parser.AstValue.setValue(AstValue.java:254)
        at com.sun.el.ValueExpressionImpl.setValue(ValueExpressionImpl.java:294)
        at org.jboss.weld.module.web.el.WeldValueExpression.setValue(WeldValueExpression.java:64)
        at org.jboss.weld.module.web.el.WeldValueExpression.setValue(WeldValueExpression.java:64)
        at com.sun.faces.facelets.el.TagValueExpression.setValue(TagValueExpression.java:131)
        ... 63 more
Caused by: java.lang.NullPointerException
        at ch.puzzle.itc.mobiliar.presentation.common.context.SessionContext.setContextId(SessionContext.java:111)
        at ch.puzzle.itc.mobiliar.presentation.common.context.SessionContext$Proxy$_$$_WeldClientProxy.setContextId(Unknown Source)
        at ch.puzzle.itc.mobiliar.presentation.templateEdit.EditTemplateView.setContextIdViewParam(EditTemplateView.java:109)
        at sun.reflect.GeneratedMethodAccessor1062.invoke(Unknown Source) [:1.8.0_151]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_151]
        at java.lang.reflect.Method.invoke(Method.java:498) [rt.jar:1.8.0_151]
        at javax.el.BeanELResolver.setValue(BeanELResolver.java:320)
        ... 70 more
```

--- a/AMW_web/src/main/java/ch/puzzle/itc/mobiliar/presentation/common/context/SessionContext.java
+++ b/AMW_web/src/main/java/ch/puzzle/itc/mobiliar/presentation/common/context/SessionContext.java
@@ -54,14 +54,15 @@ public class SessionContext implements Serializable {
     private CommonDomainService commonService;
 
     @Inject
-    Event<ContextEntity> contextEntityEvent;
+    private Event<ContextEntity> contextEntityEvent;
 
-    ContextEntity currentContext;
+    private ContextEntity currentContext;
 
-    ContextEntity globalContext;
+    private ContextEntity globalContext;
 
-    @Produces @Selected
-    public ContextEntity getCurrentContext(){
+    @Produces
+    @Selected
+    public ContextEntity getCurrentContext() {
         return currentContext;
     }
 
@@ -81,18 +82,17 @@ public class SessionContext implements Serializable {
     }
 
     private void initializeCurrentContext() {
-        if (currentContext == null){
+        if (currentContext == null) {
             currentContext = globalContext;
         }
     }
 
     private void loadContexts() {
-            contexts = Collections.unmodifiableList(contextLocator.getAllEnvironments());
+        contexts = Collections.unmodifiableList(contextLocator.getAllEnvironments());
     }
 
     private void loadTargetPlatforms() {
         targetPlatforms = Collections.unmodifiableList(commonService.getRuntimeResourceGroups());
-
     }
 
     public Integer getGlobalContextId() {
@@ -108,13 +108,12 @@ public class SessionContext implements Serializable {
     }
 
     public void setContextId(Integer contextId) {
-        if (!contextId.equals(this.contextId)) {
+        if (contextId != null && !contextId.equals(this.contextId)) {
             this.contextId = contextId;
             currentContext = findCurrentContext();
             contextEntityEvent.fire(currentContext);
         }
     }
-
 
     public List<ContextEntity> getChildrenForContext(Integer id) {
         if (id == null || id.equals(0)) {
@@ -143,8 +142,7 @@ public class SessionContext implements Serializable {
             for (ContextEntity c : contexts) {
                 if (c.getId().equals(contextId)) {
                     return c;
-                }
-                else if (c.getId().equals(getGlobalContextId())) {
+                } else if (c.getId().equals(getGlobalContextId())) {
                     global = c;
                 }
             }
@@ -153,8 +151,8 @@ public class SessionContext implements Serializable {
         return null;
     }
 
-    public boolean isEnvironment(){
-        return currentContext!=null && currentContext.isEnvironment();
+    public boolean isEnvironment() {
+        return currentContext != null && currentContext.isEnvironment();
     }
 
 }

